### PR TITLE
Gameplay page UI Design

### PR DESCRIPTION
**Pull request by cgu2020**
### Summary 

This pull request is the first step towards implementing gameplay page 

- [ ] UI designs
- [ ] Dialog Boxes

These are the general elements of the Gameplay Page subject to be changed in the release. Backend is not connected and will be done in a separate PR.

Screenshot 2024-03-05 at 8 00 07 PM

### CODE REVIEW

**Review by nit0107 — COMMENTED** [review]
Nice job updating the gameplay view and adding frontend features!

File: game/lib/gameplay/gameplay_page.dart [hunk 1]
```diff
@@ -2,9 +2,11 @@ import 'package:flutter/material.dart';
 import 'package:flutter/src/foundation/key.dart';
 import 'package:flutter/src/widgets/framework.dart';
 import 'package:flutter/src/widgets/placeholder.dart';
+import 'package:velocity_x/velocity_x.dart';
 
 import 'gameplay_map.dart';
 
+// Returns widget that fills based on tasks and tasks_completed.
 List<Widget> generate_progress(tasks, tasks_completed) {
```
Comment (neketka) [inline game/lib/gameplay/gameplay_page.dart:10]: Make sure to use camelCase when writing out variable names. Also, explicitly specify the type of the arguments so it isn't dynamic.

File: game/lib/gameplay/gameplay_page.dart [hunk 2]
```diff
@@ -2,9 +2,11 @@ import 'package:flutter/material.dart';
...
   List<Widget> progress_blocks = [];
```
Comment (neketka) [inline game/lib/gameplay/gameplay_page.dart:11]: camelCase

File: game/lib/gameplay/gameplay_page.dart [hunk 3]
```diff
@@ -2,9 +2,11 @@ import 'package:flutter/material.dart';
...
   for (var i = 0; i < tasks; i++) {
```
Comment (neketka) [inline game/lib/gameplay/gameplay_page.dart:12]: Use of tasks here is confusing, an explicit type would be helpful. Consider changing to something like taskCount.

File: game/lib/gameplay/gameplay_page.dart [hunk 4]
```diff
@@ -36,84 +38,96 @@ class _GameplayPageState extends State<GameplayPage> {
...
+                      ),
+                    ),
+                  ]),
                 ),
                 Container(
                   height: 100,
                   width: 500,
                   decoration: BoxDecoration(
-                    color: Color.fromARGB(255, 238, 238, 238),
+                    color: Color.fromARGB(255, 255, 255, 255),
```
Comment (neketka) [inline game/lib/gameplay/gameplay_page.dart:101]: Probably use a hex color here for easy copying from Figma.

**Review by neketka — CHANGES_REQUESTED** [review]
Looking good so far! Some minor nits in the comments.

**Comment by cathli66** [comment]
No need to deal with google maps integration! I'm handling that part, just let me know when the frontend is ready and we can merge in the PR so I can add in the map. You could make a separate PR for connecting it to the backend after

**Review by cathli66 — APPROVED** [review]
All the UI stuff looks good to me! I guess you must have updated or reinstalled the dependencies to change package-lock.json? Weird that it would be completely rewritten

**Review by cathli66 — DISMISSED** [review]

**Review by cathli66 — APPROVED** [review]
The UI matches the Figma designs!

**Pull request merged**